### PR TITLE
Support Object configuration parameters

### PR DESCRIPTION
**Pull request by jeremyjliu**
Adds support for Object configuration parameters, with named instance fields of configurable Parameter types.

### CODE REVIEW

**Comment by jeremyjliu** [comment]
A comment: I saw a lot of parallels with ObjectParameter and the Entity class, but it would be a large undertaking to merge the Entity and Parameter concepts so I opted to put this together for now.

File: conda/common/configuration.py [hunk 1]
```diff
@@ -709,14 +728,6 @@ def __init__(self, name, value, element_type, key_flag, value_flags, validation=
...
             name, value, key_flag, value_flags, validation)
 
-    def __eq__(self, other):
-        if type(other) is type(self):
-            return self.value == other.value
-        return False
-
-    def __hash__(self):
-        return hash(self.value)
-
```
Comment (jjhelmus) [inline conda/common/configuration.py]: Are these needed for comparison elsewhere?

Comment (jeremyjliu) [reply conda/common/configuration.py]: remove it bc it is the same as the superclass

**Comment by jeremyjliu** [comment]
@jjhelmus any other thoughts?

**Comment by jjhelmus** [comment]
LGTM

**Pull request merged**